$MirOS: ports/shells/zsh/patches/patch-configure_ac,v 1.1 2009/09/06 16:31:13 tg Exp $
$OpenBSD: patch-configure_ac,v 1.1 2010/09/06 18:53:51 nicm Exp $
--- configure.ac.orig	Thu Sep 23 09:41:41 2010
+++ configure.ac	Tue Mar  8 17:18:24 2011
@@ -409,7 +409,7 @@ if test -n "$auto_cflags" && test ."$ans
 fi
 if test -n "$auto_ldflags"; then
   case "${enable_zsh_debug}$host_os" in
-    yesaix*|yeshpux*|yesnetbsd*|yesopenbsd*) ;;  # "ld -g" is not valid on these systems
+    yesaix*|yeshpux*|yesmirbsd*|yesnetbsd*|yesopenbsd*) ;;  # "ld -g" is not valid on these systems
     darwin*) LDFLAGS=-Wl,-x ;;
     yes*)    LDFLAGS=-g ;;
     *)       LDFLAGS=-s ;;
@@ -702,7 +702,7 @@ AH_TEMPLATE([ZSH_NO_XOPEN],
 AC_CACHE_CHECK(if _XOPEN_SOURCE_EXTENDED should not be defined,
 zsh_cv_no_xopen,
 [[case "$host_os" in
-  *openbsd*|*freebsd5*|*freebsd6.[012]*|*aix*)
+  *freebsd5*|*freebsd6.[012]*|*aix*)
   zsh_cv_no_xopen=yes
   ;;
   *)
@@ -2646,6 +2646,7 @@ char *argv[];
     solaris*|sysv4*|esix*) DLLDFLAGS="${DLLDFLAGS=-G}" ;;
     darwin*)      DLLDFLAGS="${DLLDFLAGS=-bundle -flat_namespace -undefined suppress}" ;;
     beos*|haiku*) DLLDFLAGS="${DLLDFLAGS=-nostart}" ;;
+    mirbsd*)      DLLDFLAGS="${DLLDFLAGS=-shared -fPIC}" ;;
     openbsd*)
       if test x$zsh_cv_sys_elf = xyes; then
 	DLLDFLAGS="${DLLDFLAGS=-shared -fPIC}"
@@ -2663,6 +2664,7 @@ char *argv[];
   esac
   case "$host" in
     *-hpux*)  EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-Wl,-E}" ;;
+    *mirbsd*) EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-Wl,-E}" ;;
     *openbsd*)
        if test x$zsh_cv_sys_elf = xyes; then
 	 EXTRA_LDFLAGS="${EXTRA_LDFLAGS=-Wl,-E}"
